src/range_query/sqrt_decomp.rs

```diff
@@ -129,14 +131,54 @@ impl PiecewiseLinearFn {
     fn update_envelope(&mut self) {
         self.recent_lines.extend(self.sorted_lines.drain(..));
         self.recent_lines.sort_unstable();
-        for (slope, intercept) in self.recent_lines.drain(..) {
-            // TODO: do convex hull trick algorithm
-            self.sorted_lines.push((slope, intercept));
+        self.intersections.clear();
+
+        'outer: for (m1, b1) in self.recent_lines.drain(..) {
+            while let Some(&(m2, b2)) = self.sorted_lines.last() {
+                // The lines are sorted by intercept so an incoming line with
+                // the same slope can never beat the previous line.
+                if m2 == m1 {
+                    continue 'outer;
```

Comment:
Is `continue 'outer` equivalent to just `break`?

Reply:
"break" would execute the "sorted_lines.push(...)". I could move that statement into the loop but then I would need to manually add it in in the case that the "while" loop never runs (when there are no lines in the envelope yet).